src/filter_finder.py

```diff
@@ -0,0 +1,208 @@
+"""
+This class is a diagnostic tool used for the manual determination of what color 
+ranges can be used to threshold out all pixels corresponding to a specific
+semantic features in videos (or snippets of videos) in which a single thing is 
+filmed under controlled, static lighting conditions.
+"""
+
+import os.path
+import cv2
+import numpy as np
+
+__author__ = "Nick"
+
+class FilterFinder:
+
+	# TODO: add some basic validation and related exception handling (?)
+	def __init__(self, unknown_filters = None, known_filters = None):
+		# Increments by which values can be changed through keyboard controls
+		self.increments = [1, 5]
+
+		# quadruples of (control key, H/S/V channel index, sign of increment/decrement,
+		# index of increment to use
+		self.lower_controls = [	('a', 0, 1, 0),
+								('s', 1, 1, 0),
+								('d', 2, 1, 0),
+								('z', 0, -1, 0),
+								('x', 1, -1, 0),
+								('c', 2, -1, 0),
+
+								('A', 0, 1, 1),
+								('S', 1, 1, 1),
+								('D', 2, 1, 1),
+								('Z', 0, -1, 1),
+								('X', 1, -1, 1),
+								('C', 2, -1, 1)]
+
+		self.upper_controls = [	('g', 0, 1, 0),
+								('h', 1, 1, 0),
+								('j', 2, 1, 0),
+								('b', 0, -1, 0),
+								('n', 1, -1, 0),
+								('m', 2, -1, 0),
+
+								('G', 0, 1, 1),
+								('H', 1, 1, 1),
+								('J', 2, 1, 1),
+								('B', 0, -1, 1),
+								('N', 1, -1, 1),
+								('M', 2, -1, 1)]
+		
+		# initialize filters 
+		self.filters = {}
+		if unknown_filters is not None:
+			self.filters = self.get_filter_map(unknown_filters)
+		if known_filters is not None:
+			self.set_filters(known_filters)
+		# if both of the above conditions failed, use generic filter names
+		if self.filters == {}:
+			self.filters = self.get_filter_map("filter1", "filter2")
+
+	# display video at path and allow user to alter chosen color range live
+	def find_filter(self, path, chosen, downsize_scale = 2, write = True):
+		# if path leads to valid file
+		if os.path.isfile("../videos/Board_trim1.mp4"):
+			# if desired parameter exists in filtering parameter set
+			if chosen in self.filters:
+				# decide which color filter bounds to alter using controls based on supplied name
+				lower_chosen, upper_chosen = self.filters[chosen][0], self.filters[chosen][1]
+				
+				# try to read file
+				cap = cv2.VideoCapture(path)
+				success, frame = cap.read()
+				
+				if success:
+					# get dimensions of this video
+					height, width, _ = frame.shape
+					scaled_height, scaled_width = height/downsize_scale, width/downsize_scale
+
+					# parameters for displaying filter parameters on extracted frames
+					font = cv2.FONT_HERSHEY_SIMPLEX
```

Comment:
HELL YEA THAT FONT

Reply:
`>unironically suggesting there are other fonts`